testfiles-gd/support/pgfgd-regression-test.tex

```diff
@@ -1,18 +1,5 @@
 \input regression-test.tex
 
-\catcode`\@=11 % \makeatletter
-
-% this macro should be used _after_ graphdrawing library is loaded
-\def\pgfgdBeforeBeginDocument{
```

Comment:
Now the test fails with
```
pgfgd-debug.lua:1: attempt to index a nil value (global 'pgf')
```
because when `pgfgd-debug.lua` is read in, the whole `pgf` is not loaded yet. This explains why the `\pgfgdBeforeBeginDocument` was introduced (and used after package loading and just before `\begin{document}`). For latex format only, `\AtBeginDocument` suffices but I more or less want(ed) a portable solution (not a strong opinion).